tests/test_BlkDiagMatrix.py

```diff
@@ -332,119 +108,33 @@ def testBlkDiagMatrixSub(self):
-        mean_coeff = np.array([
-            [ 4.53531036e-04],
-            [ 2.29341625e-04],
-            [ 9.65040155e-05],
-            [-1.19442447e-04],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00]
-        ])
-
-        result = np.array([
-            [-2.81508398e-04],
-            [-1.57256547e-04],
-            [ 3.34520158e-05],
-            [ 1.44760391e-04],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [ 0.00000000e+00],
-            [-0.00000000e+00],
-            [-0.00000000e+00],
-            [-0.00000000e+00],
-            [-0.00000000e+00],
-            [-0.00000000e+00],
-            [-0.00000000e+00]])
-
-        blk_c = self.blk_a.apply(mean_coeff)
-        self.allallfunc(result, blk_c)
+        m = np.sum(self.blk_a.partition[:, 1])
+        k = 3
+        coeffm = np.arange(k * m).reshape(m, k).astype(self.blk_a.dtype)
+
+        # Manually compute
+        ind = 0
+        res = np.empty_like(coeffm)
+        for b, blk in enumerate(self.blk_a):
+            col = self.blk_a.partition[b, 1]
+            res[ind: ind + col, :] = blk @ coeffm[ind: ind + col, :]
+            ind += col
+
+        # Check ndim 1 case
+        c = self.blk_a.apply(coeffm[:, 0])
+        self.allallfunc(c, res[:, 0])
+
+        # Check ndim 2 case
+        d = self.blk_a.apply(coeffm)
+        self.allallfunc(res, d)
+
+        e = np.empty((m, k))
```

Comment:
This one's a little tricky. Needs a short comment.

Reply:
Added comment, and removed the addition (wasn't adding anything to the test).

Reply:
Ah I see…